File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/StatisticsSample/src/com/hp/workpath/sample/statisticsample/fragment/StatisticFragment.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.statisticsample.fragment;

import static com.hp.workpath.sample.statisticsample.StatisticActivity.INDEX;
import static com.hp.workpath.sample.statisticsample.StatisticActivity.LAST_JOB;

import android.os.Bundle;
import android.text.TextUtils;
import android.text.method.ScrollingMovementMethod;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.hp.workpath.api.Result;
import com.hp.workpath.api.statistics.StatisticsJobData;
import com.hp.workpath.sample.statisticsample.Logger;
import com.hp.workpath.sample.statisticsample.R;
import com.hp.workpath.sample.statisticsample.task.JobInfoTask;
import com.hp.workpath.sample.statisticsample.task.LastJobInfoTask;
import com.hp.workpath.sample.statisticsample.view.StatisticJobInfoView;

import java.util.List;

public class StatisticFragment extends Fragment {

    TextView mStatisticRawDataTextView;
    ProgressBar mProgressBar;

    StatisticsJobData mStatisticJobData;
    StatisticJobInfoView statisticView;

    int mIndex = 0;
    boolean isLastJob = false;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_statistic, container, false);
        mStatisticRawDataTextView = view.findViewById(R.id.statisticRawDataTextView);
        mStatisticRawDataTextView.setMovementMethod(new ScrollingMovementMethod());
        mProgressBar = view.findViewById(R.id.progressBar);

        statisticView = new StatisticJobInfoView(inflater, view);
        isLastJob = getArguments().getBoolean(LAST_JOB);
        if (isLastJob) {
            getLastJobInfo();
        } else {
            mIndex = getArguments().getInt(INDEX);
            getJobInfoList();
        }
        return view;
    }

    private void setRawData(StatisticsJobData statisticsJobData) {
        Gson gson = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
        mStatisticRawDataTextView.setText(gson.toJson(statisticsJobData));
    }

    private void getJobInfoList() {
        mProgressBar.setVisibility(View.VISIBLE);
        new JobInfoTask(getContext(), statisticsInterface).taskExecute(mIndex);
    }

    private void getLastJobInfo() {
        mProgressBar.setVisibility(View.VISIBLE);
        new LastJobInfoTask(getContext(), statisticsInterface).taskExecute();
    }

    private void setStatisticsJobData(StatisticsJobData statisticsJobData) {
        statisticView.setStatisticJobData(statisticsJobData);
    }

    private void showStatisticJobData() {
        if (mStatisticJobData != null) {
            setRawData(mStatisticJobData);
            setStatisticsJobData(mStatisticJobData);
        } else {
            mStatisticRawDataTextView.setText("There is no statistic information");
        }
    }

    ResponseInterface statisticsInterface = new ResponseInterface() {
        @Override
        public void success(List<StatisticsJobData> info) {
            mProgressBar.setVisibility(View.GONE);
            if (info != null && info.size() > 0) {
                mStatisticJobData = info.get(0);
                if (TextUtils.isEmpty(mStatisticJobData.getJobId())) {
                    mStatisticRawDataTextView.setText("There is no statistic information");
                } else {
                    showStatisticJobData();
                }
            } else {
                mStatisticRawDataTextView.setText("There is no statistic information");
            }
        }

        @Override
        public void failure(String msg, Result result) {
            mProgressBar.setVisibility(View.GONE);
            Logger.showResult(getActivity(), msg, result);
        }
    };
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/StatisticsSample/src/com/hp/workpath/sample/statisticsample/fragment/StatisticFragment.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.statisticsample.fragment

import android.os.Bundle
import android.text.TextUtils
import android.text.method.ScrollingMovementMethod
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.lifecycleScope
import com.google.gson.GsonBuilder
import com.hp.workpath.api.Result
import com.hp.workpath.api.statistics.StatisticsJobData
import com.hp.workpath.sample.statisticsample.Logger
import com.hp.workpath.sample.statisticsample.StatisticActivity.Companion.INDEX
import com.hp.workpath.sample.statisticsample.StatisticActivity.Companion.LAST_JOB
import com.hp.workpath.sample.statisticsample.databinding.FragmentStatisticBinding
import com.hp.workpath.sample.statisticsample.task.JobInfoTask
import com.hp.workpath.sample.statisticsample.task.LastJobInfoTask
import com.hp.workpath.sample.statisticsample.view.StatisticJobInfoView
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch


class StatisticFragment : Fragment() {

    private lateinit var statisticView: StatisticJobInfoView
    private var mStatisticJobData: StatisticsJobData? = null

    private var mBindingFragment: FragmentStatisticBinding? = null
    private val mBindingStatisticFragment get() = mBindingFragment!!
    private var mIndex = 0
    private var isLastJob = false

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // val view = inflater.inflate(R.layout.fragment_statistic, container, false)
        mBindingFragment = FragmentStatisticBinding.inflate(inflater, container, false)
        statisticView = StatisticJobInfoView(inflater, mBindingStatisticFragment.root)
        return mBindingStatisticFragment.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        mBindingStatisticFragment.statisticRawDataTextView.movementMethod =
            ScrollingMovementMethod()
        arguments?.run {
            isLastJob = getBoolean(LAST_JOB)
            if (isLastJob) {
                getLastJobInfo()
            } else {
                mIndex = getInt(INDEX)
                getJobInfoList()
            }
        }
    }

    private fun setRawData(statisticsJobData: StatisticsJobData) {
        val gson = GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create()
        mBindingStatisticFragment.statisticRawDataTextView.text = gson.toJson(statisticsJobData)
    }

    private fun getJobInfoList() {
        mBindingStatisticFragment.progressBar.visibility = View.VISIBLE
        lifecycleScope.launch(Dispatchers.Default) {
            context?.let { JobInfoTask(it, statisticsInterface).execute(mIndex) }
        }
    }

    private fun getLastJobInfo() {
        mBindingStatisticFragment.progressBar.visibility = View.VISIBLE
        lifecycleScope.launch(Dispatchers.Default) {
            context?.let { LastJobInfoTask(it, statisticsInterface).execute() }
        }
    }

    private fun setStatisticsJobData(statisticsJobData: StatisticsJobData) {
        statisticView.setStatisticJobData(statisticsJobData)
    }

    private fun showStatisticJobData() {
        mStatisticJobData?.run {
            setRawData(this)
            setStatisticsJobData(this)
        } ?: run {
            mBindingStatisticFragment.statisticRawDataTextView.text =
                "There is no statistic information"
        }
    }

    private var statisticsInterface: ResponseInterface = object : ResponseInterface {
        override fun success(info: List<StatisticsJobData?>) {
            mBindingStatisticFragment.progressBar.visibility = View.GONE
            if (info.isNotEmpty()) {
                mStatisticJobData = info[0]
                if (TextUtils.isEmpty(mStatisticJobData?.jobId)) {
                    mBindingStatisticFragment.statisticRawDataTextView.text =
                        "There is no statistic information"
                } else {
                    showStatisticJobData()
                }
            } else {
                mBindingStatisticFragment.statisticRawDataTextView.text =
                    "There is no statistic information"
            }
        }

        override fun failure(msg: String?, result: Result?) {
            mBindingStatisticFragment.progressBar.visibility = View.GONE
            Logger.showResult(activity, msg, result)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        mBindingFragment = null
    }
}